addons/FlashCards/src/presenter.js

```diff
@@ -0,0 +1,468 @@
+function AddonFlashcards_create(){
+			
+    var presenter = function(){}
+
+    presenter.configuration = {
+        isVisible: false,
+        currentCard: 1,
+        NoLoop: false,
+        IsActivity: false,
+        Favourites: false
+    };
+
+    presenter.state = {
+		isVisible: false,
+        currentCard: 1,
+        totalCards: 1,
+        NoLoop: false,
+        IsActivity: false,
+        Favourites: false,
+        ShowOnlyFavourites: false,
+        cardsScore: null,
+        cardsFavourites: null
+	};
+
+    presenter.setPlayerController = function(controller) {
+        presenter.playerController = controller;
+        presenter.eventBus = presenter.playerController.getEventBus();
+    };
+
+    presenter.createEventData = function(i,v,s) {
+        return {
+            source : presenter.configuration.addonID,
+            item: i,
+            value: v,
+            score: s
+        };
+    };
+
+	presenter.triggerEvent = function(i,v,s) {
+        var eventData = presenter.createEventData(i,v,s);
+        presenter.eventBus.sendEvent('ValueChanged', eventData);
+    };
+
+    presenter.validateModel = function(model) {
+		return {
+			isValid: true,
+			isVisible: ModelValidationUtils.validateBoolean(model["Is Visible"]),
+            NoLoop: ModelValidationUtils.validateBoolean(model['NoLoop']),
+			Favourites: ModelValidationUtils.validateBoolean(model['Favourites']),
+			HidePrevNext: ModelValidationUtils.validateBoolean(model['HidePrevNext']),
+            ShowButtons: ModelValidationUtils.validateBoolean(model['ShowButtons']),
+			IsActivity: ModelValidationUtils.validateBoolean(model["IsActivity"]),
+            currentCard: presenter.configuration.currentCard,
+            cardsScore: [],
+            cardsFavourites: [],
+			addonID: model['ID']
+		}
+	};
+
+    presenter.init = function(view, model){
+        var validatedModel = presenter.validateModel(model);
+        presenter.configuration = validatedModel;
+        presenter.Cards = model.Cards;
+        presenter.state.isVisible = presenter.configuration.isVisible;
+        presenter.state.NoLoop = presenter.configuration.NoLoop;
+        presenter.state.cardsScore = presenter.configuration.cardsScore;
+        presenter.state.cardsFavourites = presenter.configuration.cardsFavourites;
+
+        //console.log(model.ID+" init, config:");
+        //console.log(presenter.configuration);
+
+        presenter.view = view;
+        presenter.$view = $(view);
+        presenter.model = model;
+        presenter.$card = $(presenter.$view.find(".flashcards-card").get(0));
+
+        $(presenter.Cards).each(function(k,v){

```

Comment:
Prosiłbym o bardziej zrozumiałe nazwy dla argumentów metod. 

v nie jest wykorzystywane. Jeżeli aktualny element w each nie jest ci potrzebny, to ten argument funkcji może zostać pominięty.